src/borg/cache.py

```diff
@@ -612,15 +613,12 @@ def create_master_idx(chunk_idx):
             master_index_capacity = int(len(self.repository) / ChunkIndex.MAX_LOAD_FACTOR)
             if archive_ids:
                 chunk_idx = None
-                if self.progress:
-                    pi = ProgressIndicatorPercent(total=len(archive_ids), step=0.1,
-                                                  msg='%3.0f%% Syncing chunks cache. Processing archive %s',
-                                                  msgid='cache.sync')
+                pi = ProgressIndicatorPercent(total=len(archive_ids), step=0.1,
+                                              msg='%3.0f%% Syncing chunks cache. Processing archive %s',
+                                              msgid='cache.sync')
                 archive_ids_to_names = get_archive_ids_to_names(archive_ids)
-                for archive_id in archive_ids:
-                    archive_name = archive_ids_to_names.pop(archive_id)
-                    if self.progress:
-                        pi.show(info=[remove_surrogates(archive_name)])
+                for archive_id, archive_name in archive_ids_to_names.items():
```

Comment:
Actually it's the other way around. Previously it was random, and for <Py3.6 it still is. For Py3.6+ on the other hand, dict() preserves insertion order, and the loop get_archive_ids_to_names operates on the Archives, which are not explicitly sorted. However, when writing the Manifest, StableDict is used, so the entries in the msgpackd-Manifest are sorted by name. This is preserved when loading it. So for Py3.6+ this now processes in sorted-by-name order, which is probably best: `date-backupset` => clustered by date, `backupset-date` (--prefix compliance) => clustered by backupset.